KeLiu-Kryie/COMSW4156-Drop_Table_Team
# Backend server&translation

Created `Server` and `Translator` classes based on my Crow server code & Ke's translator code. Updated `main` to interface with `Server` as opposed to the database directly.

## How to test
1. Run `make run`
2. Visit [http://localhost:18080/translate/?tbt=hello&tl=jp&fl=en](http://localhost:18080/translate/?tbt=hello&tl=jp&fl=en) to see it in action

## Issues
I'm still having problems building the MongoDB-related files, but I'm working through them. For now, I just commented everything out from the DatabaseHandler.h/cpp files  -- maybe someone who has them working could integrate them into `Server` ? To support this, I excluded the database-specific link flags from the makefile, but these should be re-introduced once it's working again.

I'm also a little concerned with the translation API we're using. According to its documentation, it only supports Chinese, English, Japanese, Korean, Russian, Vietnamese and Thai. Most of these don't use the Roman alphabet (ie English letters), so we may have to add a level of interpretation.
